clang/runtime/dpct-rt/include/dpct/kernel.hpp

```diff
@@ -400,6 +401,8 @@ class kernel_function {
     ptr(q, range, a, args, extra);
   }
 
+  operator uint64_t() const { return (uint64_t)this; }
```

Comment:
Do we also need a constructor which accepts uint64_t type argument?

Reply:
No becasue CUfuntion is a pointer type to struct CUfunc_st and `nvcc` don't allow assigment of the following type.
```
CUfuntion func;
u64 value;
func = value;
```
Note: `nvcc do allow the assignment other way round, like `value = func`.